cmd/monitor.go

```diff
@@ -0,0 +1,242 @@
+			} else {
+				ds.Progress = fmt.Sprintf("%s done.", params.filePath)
+			}
+		}
+	}()
+
+	f, err := os.Open(params.filePath)
+	if err != nil {
+		return fmt.Sprintf("%s", err)
+	}
+	defer f.Close()
+
+	item, err := s.scanner.MonitorUploadFilename(f, params.remotePath, progressCh)
+	if err != nil {
+		return fmt.Sprintf("%s", err)
+	}
+
+	return fmt.Sprintf("%s %s", params.filePath, item.ID())
+}
+
+var monitorItemUploadCmdHelp = `Upload a file or files contained in a folder.
+
+This command receives one file or folder path and uploads them to your
+VirusTotal Monitor account. It returns uploaded the file paths followed by their
+corresponding monitor ID.
+You can use the "vt monitor items [monitor_id]" command for retrieving
+information about the it.`
+
+var monitorItemUploadCmdExample = `  vt monitor item upload foo.exe /remote_folder/foo.exe
+  vt monitor item upload myfolder/ /another_remote_folder/`
+
+// NewMonitorItemsUploadCmd returns a new instance of the 'mointor upload file' command.
+func NewMonitorItemsUploadCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "upload [file/folder] [remote_path]",
+		Short:   "Upload one or more files to your account",
+		Long:    monitorItemUploadCmdHelp,
+		Example: monitorItemUploadCmdExample,
+		Args:    cobra.ExactArgs(2),
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			localPath := args[0]
+			remotePath := args[1]
+
+			// Check if first arg is a file or a folder
+			pathStat, err := os.Stat(localPath)
+			if err != nil {
+				return err
+			}
+
+			ch := make(chan interface{})
+
+			switch mode := pathStat.Mode(); {
+			case mode.IsDir():
+				// Upload tree to remote
+				localPathClean := strings.TrimRight(localPath, "/") + "/"
+				remotePathClean := strings.TrimRight(remotePath, "/") + "/"
+
+				filesParams := make([]uploadParams, 0)
+				filepath.Walk(
+					localPathClean, func(path string, info os.FileInfo, err error) error {
+						if !info.Mode().IsRegular() {
+							return nil
+						}
+						relativePath := strings.SplitN(path, localPathClean, 2)[1]
+						remoteAbsoluteFilename := remotePathClean + relativePath
+						filesParams = append(filesParams, uploadParams{path, remoteAbsoluteFilename})
+						return nil
+					})
+
+				// Confirm user want to create those files in remote
+				fmt.Println("Following files are going to be created:")
+				for _, params := range filesParams {
+					fmt.Println(params.filePath + " -> " + params.remotePath)
+				}
+				var s string
+				fmt.Println("Confirm(y/n)?")
+				fmt.Scanln(&s)
+				if s != "y" {
+					return nil
+				}
+
+				go func() {
+					for _, params := range filesParams {
+						ch <- params
+					}
+					close(ch)
+				}()
+
+			case mode.IsRegular():
+				// Upload only one file to remote
+				go func() {
+					params := uploadParams{localPath, remotePath}
+					ch <- params
+					close(ch)
+				}()
+			default:
+				panic("Not a regular file or folder")
```

Comment:
Don't use `panic` here, just return an error. The error message will be printed to the user. The panic() function should be used in really exceptional circumstances in which you want to abort your program abruptly, it even prints a stack trace and you don't want that in a simple bad input from the user.

Reply:
resolved in 87b58c2bb762313b6239b620663a58ba321ebc35